app/commands/api/v1/reports/fetch_keywords.rb

```diff
@@ -12,18 +12,14 @@ def initialize(report_id)
         def call
           return unless report.pending?
 
-          keywords.in_groups_of(10).each do |group|
+          keyword_ids.in_groups_of(10).each do |group|
             sleep(long_delay)
```

Comment:
Using `sleep` is a generally accepted anti-pattern as it makes the code brittle (also harder to test 🙈 ). While a delay is needed, it could be backed into the job scheduling using `perform_in`: https://github.com/sidekiq/sidekiq/wiki/Scheduled-Jobs

Reply:
I understand. I indeed do not have much experience using the `sleep` yet 😁 Thanks for that suggestion!